lib: fix group case-sensitivity while configuring ipv6 address

patrasar opened the pull request:
Fixing the below 2 issues: 
Issue: #11810 

Issue 1:

frr(config)# ipv6 pim rp 1080::2 ff00::/8
frr(config)# no ipv6 pim rp 1080::2 FF00::/8
% Unable to find specified RP

Issue 2:
frr(config)# ipv6 pim rp 1080::2

frr(config)# no ipv6 pim rp 1080::2 ff00::/8
% Unable to find specified RP

frr(config)# no ipv6 pim rp 1080::2 FF00::/8
% Unable to find specified RP


Signed-off-by: Sarita Patra <saritap@vmware.com>

patrasar commented:
I am doing the changes for other CLIs which are taking IPV6 address

donaldsharp commented on pimd/pim6_cmd.c:
```diff
@@ -430,8 +430,20 @@ DEFPY (ipv6_pim_rp,
        "Group Address range to cover\n")
 {
 	const char *group_str = (gp_str) ? gp_str : "FF00::0/8";
+	int ret;
```
yeah this should be fixed in the cli parser not here and especially not with str2prefix and then prefix2str

tolower or toupper needs to be added in the cli somewhere so it can be normalized by the cli across all of frr

donaldsharp reviewed (changes requested):
Please fix the cli parser to normalize the string output for this

patrasar committed:
lib: fix group case-sensitivity while configuring ipv6 address

Problem:
Configure RP 1080::2 for group ff00::/8
Configure RP 1080::2 for group FF00::/8

Both are considered as different groups.

Root Cause:
We need to take care ipv6 hex case-sensitivity while
configuring ipv6 address.

Fix:
Convert the ipv6 address str to lower case before applying
the configuration.

Issue: 11810

Signed-off-by: Sarita Patra <saritap@vmware.com>

patrasar committed:
pim6d: Configure default group "ff00::/8""

Problem:
frr(config)# ipv6 pim rp 1080::2

frr(config)# show running-config
ipv6 pim rp 1080::2 ff00::/8

frr(config)# no ipv6 pim rp 1080::2 ff00::/8
% Unable to find specified RP

Root Cause:
The command "ipv6 pim rp 1080::2" takes group as
FF00::0/8, so "no ipv6 pim rp 1080::2 ff00::/8" wont
find the group "ff00::/8" and throws error.

Fix:
Configure default group as "ff00::/8"

Issue: 11810

Signed-off-by: Sarita Patra <saritap@vmware.com>

riw777 commented:
@frrbot autoclose in 1 month
